core/store/store.go

```diff
@@ -5,31 +5,31 @@ import dbm "github.com/cosmos/cosmos-db"
 // KVStore describes the basic interface for interacting with key-value stores.
 type KVStore interface {
 	// Get returns nil iff key doesn't exist. Panics on nil key.
-	Get(key []byte) []byte
+	Get(key []byte) ([]byte, error)
 
 	// Has checks if a key exists. Panics on nil key.
```

Comment:
I think all comments should be updated to returns an error in nil key. Otherwise lgtm.